lib/Sema/TypeCheckDecl.cpp

```diff
@@ -953,6 +953,61 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
           continue;
       }
 
+      // If the conflicting declarations have non-overlapping availability and
+      // - one throws and the other does not,
+      // - or they are initializers with different failability
+      // let them go.
+      bool isAcceptableVersionBasedChange = false;
+      {
+        auto *currentInit = dyn_cast<ConstructorDecl>(current);
+        auto *otherInit = dyn_cast<ConstructorDecl>(other);
+        if (currentInit && otherInit &&
+            ((currentInit->getFailability() == OTK_None) ^
```

Comment:
Interesting. Mentally I was considering this a disjunction of conditions and the below one a comparison of values, but if it's actually making it less clear then it's worth changing.